src/RegisterMapping.cpp

```diff
@@ -260,104 +275,51 @@ bool RegisterMapping::doesUpdate(const std::vector<uint16_t>& newValues) const
         throw std::logic_error("RegisterMapping: The value array has to be the same size, it cannot change.");
     }
 
-    bool different = false;
-    uint32_t i = 0;
-    while (i < newValues.size())
-    {
-        if (m_byteValues[i] != newValues[i])
-        {
-            different = true;
-            break;
-        }
-        ++i;
-    }
-
-    if (m_deadbandValue == 0.0)
-    {
-        return !m_isInitialized || different || !m_isValid;
-    }
-
-    if (!different)
-    {
-        return !m_isInitialized || !m_isValid;
-    }
-
-    bool significantChange = false;
-
-    switch (getOutputType())
-    {
-    case OutputType::UINT16:
+    if (!m_isInitialized)
     {
-        uint16_t currentValueUint16 = m_byteValues[0];
-        uint16_t newValueUint16 = newValues[0];
-
-        significantChange = newValueUint16 >= currentValueUint16 + m_deadbandValue ||
-                            newValueUint16 <= currentValueUint16 - m_deadbandValue;
+        return true;    // initial value
     }
-    break;
 
-    case OutputType::INT16:
+    if (m_frequencyFilterValue != std::chrono::milliseconds(0))
     {
-        int16_t currentValueInt16 = DataParsers::uint16ToInt16(m_byteValues[0]);
-        int16_t newValueInt16 = DataParsers::uint16ToInt16(newValues[0]);
+        bool frequentUpdate = std::chrono::high_resolution_clock::now() < m_lastUpdateTime + m_frequencyFilterValue;
 
-        significantChange =
-          newValueInt16 >= currentValueInt16 + m_deadbandValue || newValueInt16 <= currentValueInt16 - m_deadbandValue;
-    }
-    break;
-
-    case OutputType::UINT32:
-    {
-        uint32_t currentValueUint32;
-        uint32_t newValueUint32;
-        if (getOperationType() == OperationType::MERGE_BIG_ENDIAN)
+        if (m_deadbandValue == 0.0)
         {
-            currentValueUint32 = DataParsers::registersToUint32(m_byteValues, DataParsers::Endian::BIG);
-            newValueUint32 = DataParsers::registersToUint32(newValues, DataParsers::Endian::BIG);
+            return !frequentUpdate;    // freq filter
```

Comment:
Will this not return true even if no change in register value?
Mayle you should move the "if (!different)" check from below to above checking freq so you only check if there is actually diff in values